doc/source/whatsnew/v0.22.0.txt

```diff
@@ -96,7 +96,7 @@ Conversion
 Indexing
 ^^^^^^^^
 
--
+- Bug in ``MultiIndex`` which would assign object dtype to all-NaN levels (:issue:`17929`).
```

Comment:
can you move this to Other API Changes section. It has a bit more visibility there. ping when pushed as this lgtm.